Geometry/HGCalCommonData/interface/HGCalDDDConstants.h

```diff
@@ -86,6 +87,9 @@ class HGCalDDDConstants {
   int layerIndex(int lay, bool reco) const;
   unsigned int layers(bool reco) const;
   unsigned int layersInit(bool reco) const;
+  int layerType(int lay) const {
+    return ((hgpar_->layerType_.empty()) ? HGCalTypes::WaferCenter : hgpar_->layerType_[lay - hgpar_->firstLayer_]);
+  }
```

Comment:
CMS coding rule [4.3:](http://cms-sw.github.io/cms_coding_rules.html#4--technical-coding-rules-1)
Header ﬁles must not contain any implementation except for class templates and code to be inlined.

Add `inline` if you intend these methods to be inlined.